route_linux.go

```diff
@@ -1448,7 +1448,7 @@ type RouteGetOptions struct {
 	VrfName  string
 	SrcAddr  net.IP
 	UID      *uint32
-	Mark     int
+	Mark     uint
```

Comment:
Could we make it uint32 instead?

Reply:
I found only one usage for RouteGetOptions.Mark. It ignores all values less than or equal to 0,  so I don't think this change will have any impact.

https://github.com/vishvananda/netlink/blob/4d4ba1473f21ca0ccdecc301adbe4315db42718b/route_linux.go#L1558-L1563

Reply:
This is not about that. `RouteGetOptions ` is an exposed structure so are its members. It represent data as it is passed to the kernel via netlink route attributes. It is useful to the user to know what data the kernel expect:

https://github.com/torvalds/linux/blob/1dfe225e9af5bd3399a1dbc6a4df6a6041ff9c23/net/ipv4/route.c#L3240

```
mark = tb[RTA_MARK] ? nla_get_u32(tb[RTA_MARK]) : 0;
```

Reply:
Thank you for your review, it was my oversight earlier.

I updated the test code in [route_test.go](https://github.com/vishvananda/netlink/pull/983/files#diff-91f9c3332b2469d3f9d128f07bab664061bf5d8d35125b32690fd173b93d1be4) to use RouteGetWithOptions to retrieve routes with marks in the range 0x80000000 to 0xFFFFFFFF (e.g. 0xAFFFFFFF, 0xBFFFFFFF)
 I can successfully get routes with marks 0xAFFFFFFF or 0xBFFFFFFF.
```
sudo GOARCH=386 go test -v -run TestRouteFWMarkOption
```

I tried changing RouteGetOptions.Mark back to `int` type, the route could't be retrieved correctly.
`fwmark1` is of type `uint`, but `Mark` is of type `int`. Therefore, after type conversion, `Mark` will be less than 0.
```
...
fwmark1 := uint(0xAFFFFFFF)
...
routes, err = RouteGetWithOptions(dstIP, &RouteGetOptions{Mark: int(fwmark1)})
...
```
If `Mark` less than 0, This won't add the mark attribute to the conditions.
https://github.com/vishvananda/netlink/blob/4d4ba1473f21ca0ccdecc301adbe4315db42718b/route_linux.go#L1558-L1563

So I think it‘s necessary to change the type of `Mark` from `int` to `uint`.

Does this indirectly indicate that the kernel's mark, although of type `int`, doesn't care about the sign bit?